activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -366,10 +366,15 @@ def type_cast_calculated_value(value, type, operation = nil)
       end
     end
 
-    # TODO: refactor to allow non-string `select_values` (eg. Arel nodes).
     def select_for_count
       if select_values.present?
-        select_values.join(", ")
+        select_values.map do |field|
```

Comment:
This implementation is too coupled with Arel internals. I think we should expose an API on arel for doing it @sgrif WDYT?